log/field.go

```diff
@@ -0,0 +1,204 @@
+package log
```

Comment:
one problem with this approach is that adding a new field type breaks the visitor API by requiring an extra method to be implemented.